pkg/overlay/overlay.go

```diff
@@ -18,6 +18,40 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// Overlay type holds various configuration options for overlay
+// MountWithOptions accepts following type so its easier to specify
+// more verbose configuration for overlay mount.
+type Options struct {
+	// The Upper directory is normally writable layer in an overlay mount.
+	// Note!! : Following API does not handles escaping or validates correctness of the values
+	// passed to UpperDirOptionFragment instead API will try to pass values as is it
+	// to the `mount` command. Its user's responsibility to make sure they pre-validate
+	// these values. Invalid inputs may lead to undefined behviour.
+	// This is provided as-is, use it if it works for you, we can/will change/break that in the future.
+	// See discussion here for more context: https://github.com/containers/buildah/pull/3715#discussion_r786036959
+	// TODO: Should we address above comment and handle escaping of metacharacters like
+	// `comma`, `backslash` ,`colon` and any other specical characters
+	UpperDirOptionFragment string
```

Comment:
Nit: the `OptionFragment` suffix sounds redundant.
Should probably note that if this field is set, `WorkDirOptionFragment` should also be set.

Reply:
@nalind I was decided to name both of them as fragment as they are still just fragments to overlay's options as discussed here https://github.com/containers/buildah/pull/3715#discussion_r786734076 . I named both of them in same manner as it describes the similar nature of field to the reader of the API

Reply:
@nalind See that discussion, but I’m completely happy to leave naming/API design to you.